api/jobs/bot.py

```diff
@@ -75,69 +69,65 @@ def get_course_theia_session_count(course) -> int:
+        .count()
+    )
 
 
 def generate_report() -> str:
     """
-    Generate a report of the statusesof Anubis. The statuses are:
-        Course names and code
-        Number of active users this semester
-        Number of IDEs opened this semester
-        Number of students for each course
-        Number of IDES opened for each course
-    
+    Generate a report of the statuses of Anubis. The statuses are:
+                Course names and code
+                Number of active users this semester
+                Number of IDEs opened this semester
+                Number of students for each course
+                Number of IDES opened for each course
+
     :return: The text of the report
     """
-    report = ""
-    course_query = get_courses()
+
     # Course List
-    data = []
-    for course in course_query:
-        data.append([course.name, course.course_code])
-    report += tabulate(data, headers=["Course Name", "Course Code"])
+    courses = get_courses()
+    data = [[course.name, course.course_code] for course in courses]
+    report = tabulate(data, headers=["Course Name", "Course Code"])
 
-    report += "\n\n"
     # Number of users enrolled in at least on class this semester
-    report += "Total users enrolled in at least one course this semester\n"
+    report += "\n\nTotal users enrolled in at least one course this semester\n"
     report += f"{get_active_users_this_semester()}\n"
 
-    report += "\n"
     # Number of IDEs opened this semester
-    report += "Total IDEs opened this semseter\n"
-    report += f"{get_ides_opened_this_semester()}\n"
-
-    report += "\n"
-    # Number of students for each course
-    data = []
-    for course in course_query:
-        user_count = get_course_user_count(course)
-        data.append([course.name, course.course_code, user_count])
+    report += "\nTotal IDEs opened this semseter\n"
+    report += f"{get_ides_opened_this_semester()}\n\n"
+
+    # Number of students, IDEs for each course
+    data = [
+        [
+            course.name,
+            course.course_code,
+            get_course_user_count(course),
+            get_course_theia_session_count(course),
+        ]
+        for course in courses
+    ]
     report += tabulate(
-        data,
-        headers=["Course Name", "Course Code", "Total users enrolled"]
+        data, headers=["Course Name", "Course Code", "Users enrolled", "IDES opened"]
     )
 
-    report += "\n\n"
-    # Number of IDEs opened for each course
-    data = []
-    for course in course_query:
-        theia_count = get_course_theia_session_count(course)
-        data.append([course.name, course.course_code, theia_count])
-    report += tabulate(data, headers=["Course Name", "Course Code", "Total IDEs opened"])
-
     return report
 
 
-bot = commands.Bot(command_prefix='!')
+bot = commands.Bot(
+    command_prefix="!",
+    case_insensitive=True,
+    help_command=None,
+    owner_id=448648593314807809,
```

Comment:
Please don't add your `owner_id` here! I just checked the documentation, looks like this parameter can be omitted. If you insist to leave this here, please make it into an environment variable as what I did for the bot token.

Reply:
If it doesnt need to be there, we should probably avoid magic numbers